packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -297,8 +309,14 @@ export async function runHttpQuery(
         'Content-Type': 'application/json',
       });
     }
-    return JSON.stringify(gqlResponse);
+    return {
+      gqlResponse: JSON.stringify(gqlResponse),
+      responseInit,
+    };
   }
 
-  return JSON.stringify(responses);
+  return {
+    gqlResponse: JSON.stringify(responses),
```

Comment:
I think this might be better as graphqlResponse — do we really use "gql" much in actual APIs outside of graphql-tag?

Reply:
Love it